### DISCUSSION TITLE
WENO - Compression Ramp Problem

### DISCUSSION THREAD

**Discussion by falcaoceg** [AMReX-Combustion/PeleC#563] [Q&A] [answered]
Hello,

I am currently solving the compression ramp problem and I pretty much need to solve it using a high order WENO. I learned the basics about the code but it seems that, when I add an embedded plane (ramp) it is impossible to use WENO. Is it true? Is there any way around it? Does anybody know if it's going to be possible in the foreseeable future?

**Comment by marchdf** [comment, marked as answer]
Thanks for reaching out! Currently, in PeleC, the only numerical integration scheme that supports EB is the Method of Lines (MOL) pathway, which means you can't use WENO. To be clear:

> when I add an embedded plane (ramp) it is impossible to use WENO. Is it true? 

yes, it is not possible.

> Is there any way around it? 

no, not as the code stands 

> Does anybody know if it's going to be possible in the foreseeable future?

It will be possible in the near-ish future. It is on our TODO list to lift this restriction. It is not trivial to adapt the other numerical method for EB geometry but we absolutely want to do so soon (better numerics).

**Reply by hsitaram** [reply]
If you want to simulate just the ramp, you could use a setup without eb and angle the inflow velocity. Then you can use WENO.

**Comment by hsitaram** [comment]
If you want to simulate just the ramp, you could use a setup without eb and angle the inflow velocity. Then you can use the WENO reconstruction

**Reply by falcaoceg** [reply]
I need the horizontal length before reaching the corner and, only then, the ramp. Ideally, even a small portion with no plate in the beginning to consider the free flow reaching the plate, then developing, creating the detachment caused by the bubble and so on. Our in-house code solves this problem very well, the best option is WENOZQ (I hope the devs add this option too). But I started to look for other options mainly because our stability analysis will have to be extended to three dimensions and AMR is becoming more and more attractive. Thanks.

https://onlinelibrary.wiley.com/cms/asset/ec8f6b93-3367-4588-ba5a-c0831ca55144/ext12124-fig-0001-m.jpg

**Comment by RSuryaNarayan** [comment]
Hello all, 

I actually setup the case @hsitaram talked about without any EBs a while ago. You can find the case here: https://github.com/RSuryaNarayan/double-mach-reflection-PeleC/tree/main
